lib/bolt/applicator.rb

```diff
@@ -134,29 +134,43 @@ def apply(args, apply_body, scope)
 
       targets = @inventory.get_targets(args[0])
       ast = Puppet::Pops::Serialization::ToDataConverter.convert(apply_body, rich_data: true, symbol_to_string: true)
-      notify = proc { |_| nil }
 
-      r = @executor.log_action('apply catalog', targets) do
+      # TODO: what should the detailed message be? the manifest block is
+      # probably too long
+      r = @executor.log_action('apply catalog', 'Applying catalog', targets) do
         futures = targets.map do |target|
           Concurrent::Future.execute(executor: @pool) do
-            @executor.with_node_logging("Compiling manifest block", [target]) do
-              compile(target, ast, plan_vars)
+            @logger.debug("Compiling manifest block on #{target.uri}")
+            catalog = compile(target, ast, plan_vars)
+            @logger.debug(catalog)
+            catalog
+          end
+        end
+
+        log_proc = proc do |event|
+          if event[:type] == :node_result
+            result = event[:result]
+            # If the catalog finished there will be a message.
+            if result.message
+              status = result.ok? ? "Finished" : "Failed"
+              log.info("#{status} apply on #{result.target.host}: #{result.message}")
```

Comment:
This is a deviation from the other logging since it only logs a human readable result at "info". run_* functions all log the actual result json at info. Long term I think we need to support "format" for these and log human readable to stderr and json to disk or when format: json is specified.